File: 2020/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexes = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (indexes.containsKey(target - nums[i]))
                return new int[] {indexes.get(target - nums[i]), i};
            indexes.put(nums[i], i);
        }
        return new int[2];
    }
}

File: 2024/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexes = defaultdict(int)
        for i, n in enumerate(nums):
            if target - n in indexes:
                return [indexes[target - n], i]
            indexes[n] = i
        return []
